lib/tasks/evm_dba.rake

```diff
@@ -18,6 +18,55 @@ module EvmDba
     return false unless config['adapter'] == 'postgresql'
     return %w( 127.0.0.1 localhost ).include?(config['host']) || config['host'].blank?
   end
+
+  def self.with_options(*option_types, &block)
+    require 'trollop'
+
+    Trollop.options(EvmRakeHelper.extract_command_options) do
+      option_types.each do |type|
+        case type
+        when :db_credentials
+          opt :username,         "Username",                   :type => :string
+          opt :password,         "Password",                   :type => :string
+          opt :hostname,         "Hostname",                   :type => :string
+          opt :dbname,           "Database name",              :type => :string
+        when :local_file
+          opt :local_file,       "Destination file",           :type => :string, :required => true
+        when :remote_file
+          opt :remote_file_name, "Destination depot filename", :type => :string
+        when :splitable
+          opt :byte_count,        "Size to split files by",     :type => :string
+        when :remote_uri
+          opt :uri,              "Destination depot URI",      :type => :string, :required => true
+          opt :uri_username,     "Destination depot username", :type => :string
+          opt :uri_password,     "Destination depot password", :type => :string
+        when :exclude_table_data
+          opt :"exclude-table-data", "Tables to exclude data", :type => :strings
+        end
+      end
+      instance_exec &block if block_given?
+    end.tap { |opts| opts.delete_if { |k,v| v.nil? } }
```

Comment:
If more_core_extensions is available, this can be simplified to `opts.delete_nils`

Reply:
Hmm, okay, I will look into that.

Reply:
Done.